### ISSUE TITLE
MagGyro Processing Lib Updates

### ISSUE THREAD

**Issue by tpreynolds** [AA-CubeSat-Team/soci-gnc#118]
- [x] In the `Data Calibration` subsystem, you're selecting the same column in each of the three signals.
- [x] I think there's a logical error in the case of all 3 valid measurements. It looks like you're only ever averaging two signals -- I can demo this by inputting 3 valid signals with values [ 1,2,3 ], [ 2,3,4 ], and [ -1, -2, -3 ] ... what comes out is [ 1.5, 2.5, 3.5 ], which is the average of the first two. I should see an output of [ 2,3,4 ].
    - What's happening here is that the middle if-else block only executes a single branch on any given run .. it will check the if-statements in order, so once `if (u1)` evaluates to true, it executes that branch and ignores anything else. So you get the average of the first two signals only, not all three. You'll need to work out the correct solution here.
    - The unit test should actually be improved to catch this kind of error -- because you're sending in the same three signals, you won't be able to notice a bug like this. Try using three different values, and then compute what the result should be for each failure case.
- [x] Let's name the If-Action Subsystems something a bit more suggestive, like `ThreeValidSignals`, `TwoValidSignals` etc.
- [x] The input/output names should not be specific to the magnetometer. Instead, use a generic things like:
    - Inputs: `sensor_meas_data` and `sensor_meas_valid`
    - Outputs: `sensor_meas_body` and `sensor_valid`

**Comment by tpreynolds** [comment]
Just an FYI, you don't actually need to create two files for each of the Mag and Gyro library. We can pull the `maggyro_processing_lib` into FSW directly and then just set the parameters to be either for the Mags or the Gyros, no need for extra files.

**Comment by tpreynolds** [comment]
One more thing I noticed: in the case of three valid measurements...there's actually no way for us to currently average all three signals. Moreover, I'm not sure that I follow the logic in the if-else statement that you have there. I've added a fourth case that will average all three signals if either:
 - all signals agree within error
 - at least two pairs disagree within error, in this case there isn't much else we could do

I noticed this because the unit test actually does not provide the average of the three signals in the case when they're all at their "valid" values. Now it does -- I'll push my changes and ask you to look over them to make sure you agree.
